### DISCUSSION TITLE
System Explorer Canary on WIndows 10

### DISCUSSION THREAD

**Discussion by leland37** [winsiderss/systeminformer#2210] [Questions / Answers] [answered]
I tried right clicking on a program, Miscellaneous, Reduce Working Set which works fine in Canary on Windows 11.  However on Windows 10 it gives Access Denied every time.  I went back to the release version on Windows 10.  It seems Canary is very much only a Windows 11 application at the moment.  If you plan on supporting Windows 10 then please fix the bugs that make it impossible to use on a daily basis.  Then I will start testing again.

**Comment by Advait251206** [comment]
Just a note: Canary builds are generally expected to contain regressions and unfinished changes, so behavior differences between the stable release and Canary aren't necessarily indicative of dropped Windows 10 support.

Regarding **Reduce Working Set**, an "Access Denied" error can come from several places:

- The target process is protected or running at a higher integrity level.
- The required privilege (e.g. `SeIncreaseQuotaPrivilege`) is not enabled.
- A recent change in the Canary build altered how the operation is performed.
- A Windows 10-specific compatibility issue was introduced.

To help narrow it down, it would be useful to know:

- The exact Canary build version.
- Whether System Informer was run elevated.
- Whether the failure occurs for all processes or only specific ones.
- Whether the stable release succeeds on the same machine and process.

If the operation works on the same Windows 10 system using the stable release but consistently fails in Canary, that does sound like a regression worth reporting. A reproducible example (Windows 10 version, Canary build, target process, and any debug/error output) would make it much easier for the developers to investigate.

One thing I'd be careful about is concluding that Canary is "Windows 11 only" based on a single feature. It may simply be a bug in a code path that wasn't exercised on Windows 10 during testing.

**Comment by dmex** [comment, marked as answer]
The `Reduce Working Set ` feature was added... 17 years ago (Jan 20, 2010) and only Windows XP, Windows Vista and Windows 7 were available. See: https://github.com/winsiderss/systeminformer/commit/527cd771f7d1393452750f7cc1e2738d7b3f5a6f

Windows requires PROCESS_SET_QUOTA access and this hasn't changed for 30 years. The same code is still used today:
https://github.com/winsiderss/systeminformer/blob/9fd79433960e0d9ba721a176d95281aa1defb869/SystemInformer/actions.c#L3294-L3312
